src/main/java/org/junitpioneer/internal/PioneerUtils.java

```diff
@@ -92,12 +92,9 @@ public static Optional<Method> findMethodCurrentOrEnclosing(Class<?> clazz, Stri
 	 */
 	public static List<ExtensionContext> findAllContexts(ExtensionContext context) {
 		List<ExtensionContext> allContexts = new ArrayList<>();
-		allContexts.add(context);
-		List<ExtensionContext> parentContexts = context
-				.getParent()
-				.map(PioneerUtils::findAllContexts)
-				.orElse(List.of());
-		allContexts.addAll(parentContexts);
+		for (var c = Optional.of(context); c.isPresent(); c = c.get().getParent()) {
+			allContexts.add(c.get());
+		}
```

Comment:
Hmmm, is this really "better"? Even if one prefers the loop, I see no advantage in using `Optional` over `null`.

```suggestion
		for (var c = context; c != null; c = c.getParent()) {
			allContexts.add(c);
		}
```

Reply:
`getParent()` returns `Optional`

Reply:
But we can do:
```java
for (var c = context; c != null; c = c.getParent().orElse(null)) {
	allContexts.add(c);
}
```

Reply:
Ugh, I see. This of course changes things. Then, please feel free to use any variant you prefer. ✌️